### ISSUE TITLE
"pv" hangs example/loopback

### ISSUE THREAD

**Issue by rfjakob** [hanwen/go-fuse#52]
I am building an encrypted overlay filesystem, gocryptfs, on top of go-fuse (will release it on github soon). I noticed an issue with using pv that can also be reproduced with the loopback example:

Executing `pv > foo < /dev/zero` inside a loopback mount hangs both pv and loopback.
pv cannot be killed at that point (not even -9), trying to strace it will hang strace. The only way out is to `pkill -9 loopback`.

stracing loopback shows that all threads have been stopped:

	strace -p $(pgrep loopback) -f
	Process 22023 attached with 4 threads
	[pid 22031] --- stopped by SIGTTOU ---
	[pid 22027] --- stopped by SIGTTOU ---
	[pid 22024] --- stopped by SIGTTOU ---
	[pid 22023] --- stopped by SIGTTOU ---

However, starting pv under strace, I managed to capture this:

    strace -f pv > foo < /dev/zero
    [....]
	mmap(NULL, 135168, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8dbd74c000
	select(1, [0], [], NULL, {0, 90000})    = 1 (in [0], left {0, 89998})
	splice(0, NULL, 1, NULL, 131072, SPLICE_F_MORE) = -1 EINVAL (Invalid argument)
	read(0, "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 131072) = 131072
	select(2, [], [1], NULL, {0, 90000}
    [.....hangs....]

The interesting thing here is the "splice()" call, which seems to be an optimisation for doing-zero copy reads. I guess this is where loopback goes AWOL.

**Comment by hanwen** [comment]
I won't have time to look at this for a couple of weeks. Can you ping me next month? I think it should be fairly easy to disable the splice() functionality.

**Comment by rfjakob** [comment]
Thanks for the quick reply. I will see what I can get done myself till then!

**Comment by zhanwen** [comment]
hi,nice too meet you, also I name is hanwen.we have common name.

**Comment by rfjakob** [comment]
Ok, I think I got it. This only happens if `loopback` is backgrounded in the terminal. When it is, starting `pv` will trigger an "Unimplemented" message to stderr, which will cause the kernel to send SIGTTOU, which will stop the process.

Strace of loopback:

    [pid 31124] write(2, "2015/08/27 20:01:07 Unimplemente"..., 46 <unfinished ...>
    [...]
    [pid 31124] --- SIGTTOU {si_signo=SIGTTOU, si_code=SI_KERNEL, ...
    [...]

The obvious solution is to ignore SIGTTOU using `signal.Ignore()`. However, this function was added in Go 1.5 https://github.com/golang/go/commit/194ad16b834daf7ad8fb468845c5532f528970a6 so I don't think we can use that yet.

Workarounds for now:

- Don't background the process *OR*
- Redirect output to a file or null like `./loopback &> /dev/null`
